tests/testOutputRoot.py

```diff
@@ -65,27 +62,37 @@ class OutputRootTestCase(unittest.TestCase):
     """A test case for output roots."""
 
     def setUp(self):
-        # Note: these will succeed even if the directories don't exist.
-        subprocess.call(["rm", "-rf", testOutput])
-        subprocess.call(["rm", "-rf", testOutput2])
-        subprocess.call(["rm", "-rf", testInput1])
-        subprocess.call(["rm", "-rf", testInput2])
+        self.tempDirs = {}
 
     def tearDown(self):
-        subprocess.call(["rm", "-rf", testOutput, testOutput2,
-                         testOutput3, testInput1, testInput2])
+        for d in self.tempDirs:
+            shutil.rmtree(self.tempDirs[d])
+
+    def mkdtemp(self, prefix):
+        """Create a temporary directory and return its name.
+        The resulting path is stored in a dict (self.tempDirs) with
+        the supplied prefix as key. This allows the name to be retrieved
+        later. The directory path is returned.
+        A '-' is appended to the prefix if not there."""
+        dprefix = prefix
+        if not dprefix.endswith('-'):
+            dprefix = dprefix + '-'
+        tempDir = tempfile.mkdtemp(dir=ROOT, prefix=dprefix)
+        self.tempDirs[prefix] = tempDir
+        return tempDir
```

Comment:
this feels complex to me; I think I would have created a top-level temp folder and called that `self.testDir`, and put the output roots for each test method into that folder. But this seems to work so I'm ok if you want to leave it.